metadata-ingestion/src/datahub/ingestion/source/superset.py

```diff
@@ -494,9 +580,126 @@ def emit_chart_mces(self) -> Iterable[MetadataWorkUnit]:
                     entity_urn=chart_snapshot.urn,
                 )
 
+    def gen_schema_fields(self, column_data: List[Dict[str, str]]) -> List[SchemaField]:
+        schema_fields: List[SchemaField] = []
+        for col in column_data:
+            data_type = SUPERSET_FIELD_TYPE_MAPPINGS.get(col.get("type", ""))
+            field = SchemaField(
+                fieldPath=col.get("column_name", ""),
+                type=SchemaFieldDataType(data_type() if data_type else NullType()),
+                nativeDataType="",
+                description=col.get("column_name", ""),
+                nullable=True,
+            )
+            schema_fields.append(field)
+        return schema_fields
+
+    def gen_schema_metadata(
+        self,
+        dataset_response: dict,
+    ) -> SchemaMetadata:
+        dataset_response = dataset_response.get("result", {})
+        column_data = dataset_response.get("columns", [])
+        schema_metadata = SchemaMetadata(
+            schemaName=dataset_response.get("table_name", ""),
+            platform=make_data_platform_urn(self.platform),
+            version=0,
+            hash="",
+            platformSchema=MySqlDDL(tableSchema=""),
+            fields=self.gen_schema_fields(column_data),
+        )
+        return schema_metadata
+
+    def gen_dataset_urn(self, datahub_dataset_name: str) -> str:
+        return make_dataset_urn_with_platform_instance(
+            platform=self.platform,
+            name=datahub_dataset_name,
+            platform_instance=self.config.platform_instance,
+            env=self.config.env,
+        )
+
+    def construct_dataset_from_dataset_data(
+        self, dataset_data: dict
```

Comment:
A broader comment - I think we should start mapping the API responses into a normalized `SupersetDataset` type (either a dataclass or pydantic model) as soon as we get it back, and then use that type throughout the rest of the code

We've done this for other connectors 
- here's a StoredProcedure type for sql server https://github.com/datahub-project/datahub/blob/c3f9a9206d505d49e70b1795476c4928c87f2a42/metadata-ingestion/src/datahub/ingestion/source/sql/mssql/job_models.py#L97
- qlik data classes https://github.com/datahub-project/datahub/blob/8fa82874a478004ceeec4846cc41493de1b6afb0/metadata-ingestion/src/datahub/ingestion/source/qlik_sense/data_classes.py#L173

Overall, this results in code that's significantly easier to reason about, since parsing API responses is more clearly separated from the generation of datahub metadata

We don't need to go back and refactor the existing code (unless you want to :), but we should start doing this for the new code (e.g. datasets) moving forwards

Reply:
Totally in agreement here. I've added a very basic implementation of the `SupersetDataset`, subsequent PRs will add onto this (metric/owners)